lib/build/amodro-trace/lib/parse.js

```diff
@@ -1071,10 +1073,86 @@ define(['esprima', './lang'], function (esprima, lang) {
                 }
 
             }
-        }
+        };
 
         return result;
     };
 
+    /**
+     * Adds map entries for paths that have specific extensions (currently only .json) to two requirejs configs:
+     * - the requirejs config passed to amodrotrace (so that the same map can be sent to the browser)
+     * - the requirejs config of the current context of amodrotrace
+     * 
+     * Example: require("./my-file.json") of the "my-package/index" module will result in:
+     * {
+     *  map: {
+     *    "*": {
+     *      "my-package/my-file.json": "json!my-package/my-file.json"
+     *    }    
+     *  }
+     * }
+     */
+    parse.registerPluginPaths = function (deps, loaderConfig, parentModule, ctx) {
+        for(var i = 0; i < deps.length; i++) {
+            if (parse.hasPrefix(deps[i])) 
+                continue;
+
+            var extension = parse.getExtension(deps[i]);
+            var pluginExtensions = ['json'];
+
+            if (extension && pluginExtensions.indexOf(extension) > -1) {
+                var depMap = ctx.makeModuleMap(deps[i],
+                                ctx.makeModuleMap(parentModule),
+                                null,
+                                true);
+
+                var resolvedName = depMap.name;
+                var nameWithPluginPrefix = extension + '!' + depMap.name;
+
+                // add map to loader config that's passed to amodrotrace
+                if (!loaderConfig.map) loaderConfig.map = {};
+                if (!loaderConfig.map['*']) loaderConfig.map['*'] = {};
+                loaderConfig.map['*'][resolvedName] = nameWithPluginPrefix;
+
+                // configure RequireJS with the map
+                var config = { 
+                    map: { 
+                        "*": {}
+                    }
+                };
+
+                config.map['*'][resolvedName] = nameWithPluginPrefix;
+
+                ctx.configure(config);
+            }
+        }
+    };
+
+    /**
+     * Returns the extension of a file
+     */
+    parse.getExtension = function (path) {
+        var extensionRegex = /(?:\.)([0-9a-z]+)$/;
+        var extMatch = path.match(extensionRegex);
+
+        if (extMatch && extMatch.length > 1) {
+            return extMatch[1];
+        }
+
+        return null;
+    };
+
+    /**
+     * Returns whether or not the path has a plugin prefix
+     * E.g. json!some-file.json -> true
+     * some-file.json -> false
+     */
+    parse.hasPrefix = function (path) {
+        var index = path ? path.indexOf('!') : -1;
```

Comment:
often people, including me, coming from JSPM or the likes, put the bang as a suffix. so above example would be `some-file.json!json`. Should the hasPrefix method potentially check for the pattern `[prefix]BANG[filename]`?